baremetal/metal3datatemplate_manager_test.go

```diff
@@ -352,6 +352,7 @@ var _ = Describe("Metal3DataTemplate manager", func() {
 						Name:              "abcf",
 						Namespace:         namespaceName,
 						DeletionTimestamp: &timeNow,
+						Finalizers:        []string{"ipclaim.ipam.metal3.io"},
```

Comment:
trying to understand this addition, if its was missing how were we testing the finalizer previously.

Reply:
This change is due to [CR 0.15 related](https://github.com/kubernetes-sigs/controller-runtime/pull/2316). Now CR does not accept objects with `deletionTImeStamp` without `Finalizers`